orttraining/orttraining/core/session/training_session.h

```diff
@@ -69,8 +69,9 @@ class TrainingSession : public InferenceSession {
     DistributedConfiguration distributed_config{};
 
     struct MixedPrecisionConfiguration {
-      // Whether to use FP16 initializers.
-      bool use_fp16_initializers{};
+      // Whether to use mixed precision initializers.
+      bool use_mixed_precision_initializers{};
+      ONNX_NAMESPACE::TensorProto_DataType mixed_precision_type{ONNX_NAMESPACE::TensorProto_DataType_FLOAT16};
```

Comment:
> ONNX_NAMESPACE::TensorProto_DataType [](start = 5, length = 37)

Do we want to introduce an enum type for this? 
TensorProto_DataType would allow all types, including float, double, etc...